File: 백준/Bronze/2231. 분해합/분해합.py
n = int(input())
generator = 0 # 생성자 저장

for i in range(1, n+1):
    num = sum(map(int, str(i)))
    if i + num == n:
        generator = i
        break
if generator:
    print(generator)
else:
    print(0)

File: 백준/Bronze/2231. 분해합/분해합.kt
import java.io.*
fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val num = br.readLine().toInt()
    var generator = 0
    for(i in 1 until num){
        var sum = i
        var j = i
        while(j>0){
            sum += j%10
            j/=10
        }
        if(sum == num){
           generator = i
            break
        }
    }
    bw.write(generator.toString())
    bw.flush()
    bw.close()
    br.close()
 
}